### DISCUSSION TITLE
Slate Content Export as PDF and Word Document

### DISCUSSION THREAD

**Discussion by MujtabaInfini8ai** [ianstormtaylor/slate#5577] [Q&A]
I am using slate editor and want to export the editor content as **PDF** and **Word document**. Anyone with working solution. Because i tried so much option but didn't worked for me?

**Comment by gerardnico** [comment]
The format of the slate document is a json of nodes where the developers choose freely the type.
I'm doubtful that there is an to-go implementation of what you ask.

If you don't have any knowledge of the pdf and word format, I would recommend that you go first to html.

There is plenty of library to go from there.

https://dev.to/handdot/generate-a-pdf-in-js-summary-and-comparison-of-libraries-3k0p

**Reply by MujtabaInfini8ai** [reply]
i tried to do it with slate-html-serializer but i think its depericated....i tried to convert the slate content into html but i cant find way out

**Reply by gerardnico** [reply]
What you are asking is really basic stuff if you have a basic knowledge of slate and JavaScript.

Level up them or ask a more senior person to help you out.
You can't ask a general question as you do and expect an specific answer. No magic bullet unfortunately.

All the best.

**Reply by MujtabaInfini8ai** [reply]
Thank you for your response.
I have implemented the functionality by converting the slate content into html and passing html to download the Docx and PDF

**Comment by SpaceDandy13** [comment]
Hi @MujtabaInfini8ai,

You mentioned that you have successfully implemented this functionality by converting the Slate content to HTML, and then using that HTML to generate Docx and PDF files. I am exploring a similar solution but facing some challenges.

Would you mind sharing more details about how you achieved this? Specifically, I'm curious about how you handled the conversion from Slate content to HTML, and then how you transformed this HTML into PDF and Docx formats.

Any insights into the libraries or tools you used, code snippets, or challenges you encountered and how you overcame them would be incredibly helpful.

Thank you for your contribution to the community, and I look forward to hearing from you!

Best regards.
